### PR TITLE
feat: company profile and create job feature completed

### PR DESCRIPTION
**Pull request by sabbirrifat**
Features:
* Company Profile Creation
* Create Job with all functionalities
* Email sending functionalities with Resend
* Integration of Smart Contract
* Lots of bug fixing throughout the application 

Tasks:
https://github.com/GoodHive/app/issues/123
https://github.com/GoodHive/app/issues/137
https://github.com/GoodHive/app/issues/140
https://github.com/GoodHive/app/issues/141
https://github.com/GoodHive/app/issues/143
https://github.com/GoodHive/app/issues/146

### CODE REVIEW

**Comment by n1c01a5** [comment]
Hi @sabbirrifat 👋 can you fix the deployment ?

File: app/api/companies/create-job/route.ts [hunk 1]
```diff
@@ -43,13 +61,34 @@ export async function POST(request: Request) {
...
+    // now get the saved job id and return that
+    const savedJob = await sql`
+      SELECT id
+      FROM goodhive.job_offers
+      WHERE wallet_address = ${walletAddress}
+      ORDER BY id DESC
+      LIMIT 1
+    `;
+    const jobId = savedJob[0].id;
+    console.log("jobId", jobId);
```
Comment (n1c01a5) [inline app/api/companies/create-job/route.ts:86]: Can you remove logs

File: app/api/companies/my-profile/route.ts [hunk 2]
```diff
@@ -62,4 +89,52 @@ export async function POST(request: Request) {
...
\ No newline at end of file
+}
+
+export async function GET(request: NextRequest) {
+  const searchParamsEntries = request.nextUrl.searchParams.entries();
+  const searchParams = Object.fromEntries(searchParamsEntries);
+  
+  const { walletAddress } = searchParams;
+
+  console.log("wall address >>", walletAddress);
```
Comment (n1c01a5) [inline app/api/companies/my-profile/route.ts:100]: remove log

File: app/api/companies/my-profile/route.ts [hunk 3]
```diff
@@ -62,4 +89,52 @@ export async function POST(request: Request) {
...
+        WHERE wallet_address = ${walletAddress}
+      `;
+
+    if (company.length === 0) {
+      return new Response(JSON.stringify({ message: "Company not found" }), {
+        status: 404,
+      });
+    }
+
+    console.log("company >>", company[0]);
```
Comment (n1c01a5) [inline app/api/companies/my-profile/route.ts:130]: remove log

File: app/api/companies/update-escrow/route.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import postgres from "postgres";
+
+export async function POST(request: Request) {
+  const { id, escrowAmount } = await request.json();
+
```
Comment (n1c01a5) [inline app/api/companies/update-escrow/route.ts:5]:
```
    // Input Validation
    if (!id || typeof id !== 'number') { // Assuming 'id' is a numeric value
      return new Response(JSON.stringify({ message: "Invalid or missing job id" }), {
        status: 400,
      });
    }

    if (typeof escrowAmount !== 'number') { // Validate escrowAmount as a number
      return new Response(JSON.stringify({ message: "Invalid escrow amount" }), {
        status: 400,
      });
    }
```

**Comment by sabbirrifat** [comment]
> Hi @sabbirrifat 👋 can you fix the deployment ?

hi @n1c01a5 , I think it is just showing because of some unresolved comments in vercel but deployment is working fine

**Review by n1c01a5 — CHANGES_REQUESTED** [review]
Looks good but can you fix some little issues :

- remove console logs
- use kebab-case for all filenames

File: app/api/companies/create-job/route.ts (hunk 1 above)
Comment (sabbirrifat) [reply app/api/companies/create-job/route.ts:86]: resolved!

File: app/api/companies/my-profile/route.ts (hunk 2 above)
Comment (sabbirrifat) [reply app/api/companies/my-profile/route.ts:100]: resolved!

File: app/api/companies/my-profile/route.ts (hunk 3 above)
Comment (sabbirrifat) [reply app/api/companies/my-profile/route.ts:130]: resolved!

File: app/companies/create-job/page.tsx [hunk 5]
```diff
@@ -1,144 +1,356 @@
...
+          : data.chain === "gnosis-chain"
+          ? gnosisChainTokens[
+              gnosisChainTokens.findIndex(
+                (token) => token.value === data.currency
+              )
+            ]
+          : null
+      );
+    } catch (error) {
+      console.log(error);
```
[the remark this answers is not in the document]
Comment (sabbirrifat) [reply app/companies/create-job/page.tsx:288]: resolved!

File: app/companies/my-profile/page.tsx [hunk 6]
```diff
@@ -23,37 +44,37 @@ export default function MyProfile() {
...
-          const { imageUrl } = await postImageResponse.json();
-          setImageUrl(imageUrl);
-        } else {
-          console.error(postImageResponse.statusText);
-        }
-        setIsLoading(false);
-      };
+      if (profileResponse.ok) {
+        const profileData = await profileResponse.json();
+        console.log("profile >>", profileData);
```
[the remark this answers is not in the document]
Comment (sabbirrifat) [reply app/companies/my-profile/page.tsx:61]: resolved!

File: app/constants/countriesData.js [hunk 7]
```diff
@@ -0,0 +1,1767 @@
+export const countriesData =[
```
[the remark this answers is not in the document]
Comment (sabbirrifat) [reply app/constants/countriesData.js:1]: resolved!

File: app/api/companies/update-escrow/route.ts (hunk 4 above)
Comment (sabbirrifat) [reply app/api/companies/update-escrow/route.ts:5]: resolved!

File: app/companies/create-job/PopupModal/PopupModal.tsx [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+import { generateContent } from "./PopupModal.utils";
+import { AddressContext } from "@components/context";
+import { useCreateJob } from "@hooks/CreateJob";
+
+export const PopupModal: FC<AddFundsModalProps> = (props) => {
+  const [amount, setAmount] = useState(0);
+  const [isFullAmount, setIsFullAmount] = useState(false);
+  const [transferAddress, setTransferAddress] = useState("");
+
+  const { jobId, open, onClose, type, onSubmit } = props;
```
[the remark this answers is not in the document]
Comment (sabbirrifat) [reply app/companies/create-job/PopupModal/PopupModal.tsx:16]: @n1c01a5 Actually, I'm trying to follow a pattern to make components more readable. So even for the components that have fewer props still I'm doing it in this way. Lemme know what u think.

**Pull request merged**